ci/infra/openstack/output.tf

```diff
@@ -1,15 +1,23 @@
 output "ip_masters" {
-  value = "${zipmap(openstack_compute_instance_v2.master.*.name, openstack_networking_floatingip_v2.master_ext.*.address)}"
+  value = zipmap(
+    length(openstack_compute_instance_v2.master.*) > 0 ? openstack_compute_instance_v2.worker.*.name :
+      [for i,v in openstack_networking_floatingip_v2.master_ext.*.address: format("caasp-master-${var.stack_name}-%d", i)],
+    openstack_networking_floatingip_v2.master_ext.*.address)
 }
```

Comment:
If I understand it correctly, the patch assumes the problem is that the list of masters is empty. But I think a more general problem is that zipmap assumes both lists are the same length. It can still happen (and IIRC it actually happened) that the list of IP addresses didn't match the list of master (some ip address could not allocatted?)  Maybe checking that both lists have the same length and the length is not 0 would be a more robust check?

Reply:
umm, that is right, but the only occurrence I have seen of this is that the list of names is empty due to whatever happened behind the scenes, so Im assuming that terraform is unable to obtain the names of ALL the resources, not just one.

Do you think it could happen that terraform is able to obtain some of the names, i.e. a partial list of names?

In that case, the complexity and ugliness of this just increased tenfold, if we cant asume that it either fails to obtain everything :(

Reply:
I mean, I can write a equal check in there, but if either names or addresses are missing, what is the expected output? Do we just NOT write the output? Do we write an empty output and let the user figure it out?

This is the problem with incomplete reports, there is not much to understand from a small subset of logs, having the full logs of that failure will help us understand what was really missing and if terraform actually failed to allocate the machine (I thought a failure to allocate a resource means no output :O). That or at least a way to reproduce it :)

Reply:
We don't have logs and we cannot solve that issue. Regarding the "incompleteness" of the report, I think that this doesn't change the rationale of my comment: the root cause of the problem is that both lists should match. Checking for a) lengths match AND b) length> 0 should address any other possible case, including the only case reported.